docs/bucket-notifications/README.md

```diff
@@ -0,0 +1,435 @@
+[![Slack](https://slack.minio.io/slack?type=svg)](https://slack.minio.io)
+
+Minio server supports Amazon S3 compatible bucket event notification for following targets [AMQP](https://www.amqp.org/about/what), [Elasticsearch](https://www.elastic.co/guide/en/elasticsearch/reference/current/getting-started.html) , [Redis](http://redis.io/documentation), [nats.io](http://nats.io/) and [PostgreSQL](https://www.postgresql.org).
+
+##  Prerequisites
+
+* Install and configure Minio Server from [here](http://docs.minio.io/docs/minio).
+* Install and configure Minio Client from [here](https://docs.minio.io/docs/minio-client-quickstart-guide).
+
+| Notification Targets| 
+|:--
+| [`RabbitMQ`](#RabbitMQ) |   
+| [`Elasticsearch`](#Elasticsearch) |    
+| [`Redis`](#Redis) |    
+| [`NATS`](#NATS) |    
+| [`PostgreSQL`](#PostgreSQL) |    
+
+<a name="RabbitMQ"></a>
+## Publish Minio events via RabbitMQ 
+
+Install RabbitMQ from [here](https://www.rabbitmq.com/).
+
+### Recipe steps
+
+### Step 1: Add RabbitMQ endpoint to Minio
+
+The default location of Minio server configuration file is ``~/.minio/config.json``. Update the RabbitMQ configuration block in ``config.json`` as follows:
+
+```
+"amqp": {
+			"1": {
+				"enable": true,
+				"url": "amqp://myuser:mypassword@localhost:5672",
+				"exchange": "bucketevents",
+				"routingKey": "bucketlogs",
+				"exchangeType": "fanout",
+				"mandatory": false,
+				"immediate": false,
+				"durable": false,
+				"internal": false,
+				"noWait": false,
+				"autoDeleted": false
+			}
+        }
+```
+Restart Minio server to reflect config changes. Minio supports all the exchange available in [RabbitMQ](https://www.rabbitmq.com/). For this setup we are using ``fanout`` exchange.
+
+If you are running [distributed Minio](https://docs.minio.io/docs/distributed-minio-quickstart-guide), modify ``~/.minio/config.json`` with these local changes on all the nodes.
+
+### Step 2: Enable bucket notification using Minio client
+
+We will enable bucket events only when JPEG images are uploaded or deleted from ``images`` bucket on ``myminio`` server. Here ARN value is ``arn:minio:sqs:us-east-1:1:amqp``. To understand more about ARN please follow [AWS ARN](http://docs.aws.amazon.com/general/latest/gr/aws-arns-and-namespaces.html) documentation.
+
+```
+mc mb myminio/images
+mc events add  myminio/images arn:minio:sqs:us-east-1:1:amqp    --suffix .jpg
+mc events list myminio/images
+arn:minio:sqs:us-east-1:1:amqp s3:ObjectCreated:*,s3:ObjectRemoved:* Filter: suffix=”.jpg”
+```
+
+### Step 3: Testing on RabbitMQ
```

Comment:
Step 1 & 2 were : Add, Enable. Therefore Step 3 : Test on RabbitMQ not Testing on RabbitMQ.